### PR TITLE
Introduce subError in TokenErrorResponse.

### PR DESCRIPTION
**Pull request by rapong**
- Introduce subError field.
- Introduce bad_token subError.
- remove redundant fields from BrokerErrorResponse.
- replace redundant constant and make them use the one in AuthenticationConstants instead.

### CODE REVIEW

File: common/src/main/java/com/microsoft/identity/common/internal/platform/Device.java [hunk 1]
```diff
@@ -44,7 +44,7 @@ private Device() {
         final Map<String, String> platformParameters = new HashMap<>();
 
         platformParameters.put(PlatformIdParameters.PRODUCT, PlatformIdParameters.PRODUCT_NAME);
-        platformParameters.put(PlatformIdParameters.VERSION, "0.1.3");
+        platformParameters.put(PlatformIdParameters.VERSION, PlatformIdParameters.PRODUCT_VERSION);
```
Comment (iambmelt) [inline common/src/main/java/com/microsoft/identity/common/internal/platform/Device.java]: 👍

File: common/src/main/java/com/microsoft/identity/common/internal/platform/Device.java [hunk 2]
```diff
@@ -70,6 +70,11 @@ private Device() {
          */
         static final String PRODUCT_NAME = "MSAL.Android";
 
+        /**
+         * The String representing the sdk platform version.
+         */
+        static final String PRODUCT_VERSION = "0.2.2";
```
Comment (iambmelt) [inline common/src/main/java/com/microsoft/identity/common/internal/platform/Device.java]: Let's pull this from `BuildConfig` - this looks like something we've been forgetting to update if it's still @ `0.1.3`.

Comment (iambmelt) [reply common/src/main/java/com/microsoft/identity/common/internal/platform/Device.java]: Issue filed here: https://github.com/AzureAD/microsoft-authentication-library-common-for-android/issues/385

Comment (rapong) [reply common/src/main/java/com/microsoft/identity/common/internal/platform/Device.java]: Let's do that separately.

**Pull request merged**